### ISSUE TITLE
BUG. GameObjectContext

### ISSUE THREAD

**Issue by Lailore** [ssannandeji/Zenject-2019#353]
In main installer I have code
```csharp
public class GameInstaller : MonoInstaller {
    public GameObject CameraPrefab;

    public override void InstallBindings() {
        Container.Bind<ICamera>().FromMethod(method: CreateCamera).AsSingle();
    }

    private ICamera CreateCamera(InjectContext _) {
        var gameObjectContext = Container.InstantiatePrefabForComponent<GameObjectContext>(prefab: CameraPrefab);
        return gameObjectContext.Container.Resolve<ICamera>();
    }
}
```

My CameraPrefab have gameobject context with installer
```csharp
public class CameraInstaller : MonoInstaller {
        public override void InstallBindings() {
            Container.Bind<ICamera>().To<Camera>().AsSingle();
            Container.Bind<IPositioned>().To<ICamera>().FromResolve();
        }
    }
```
And then i get exeption
```
ZenjectException: Found circular dependency when creating type 'Lailore.Interfaces.ICamera'. Object graph: PositionChangerView
PositionChangerView
```

This gameobject installer work fine
```csharp
public class CameraInstaller : MonoInstaller {
        public override void InstallBindings() {
            Container.Bind<ICamera>().To<Camera>().AsSingle();
            Container.Bind<IPositioned>().FromMethod(method: context => Container.Resolve<ICamera>());
        }
    }
```

**Comment by svermeulen** [comment]
I just tried to reproduce but it worked for me.  Can you provide a complete minimal reproducible example showing the issue?  And also mention the zenject version you're using.

**Comment by Lailore** [comment]
Sure, and thx for fast replay.
[ZenjectBug.zip](https://github.com/modesttree/Zenject/files/1385752/ZenjectBug.zip)
Version: 5.4.0 (Oct 03, 2017) from Unity3d AssetStore

**Comment by svermeulen** [comment]
Ok thanks that was very helpful.  I can see what the issue is, but it's not clear whether it is a bug or not.  When you use FromResolve like that, it will actually bind to multiple matches.  So it will bind IBar to `Container.ResolveAll<IFoo>`.  And when you call ResolveAll it will find all matches including those in parent containers (unlike Resolve<> which will only 'fallback' on parent containers if it can't find one in the local container).

So it is kind of behaving as designed.

One fix might be to change Resolve to ResolveAll and make Resolve only match one.  That would be a breaking change but would make more sense for the api.

Another fix if you just want to get things working would be to change it to this:

```
public class GameObjectInstallerBug : MonoInstaller
{
    public override void InstallBindings()
    {
        Container.Bind(typeof(IBar), typeof(IFoo)).To<Foo>().AsSingle();
    }
}
```

Which doesn't use FromResolve so doesn't have this problem

**Comment by svermeulen** [comment]
Fixed on develop branch for 6.0.  There is now FromResolve and FromResolveAll methods to cover both cases.  Also added FromResolveAllGetter for consistency
